mm2src/mm2_main/src/lp_swap/swap_watcher.rs

```diff
@@ -181,24 +182,31 @@ impl State for ValidateTakerFee {
 
     async fn on_changed(self: Box<Self>, watcher_ctx: &mut WatcherStateMachine) -> StateResult<WatcherStateMachine> {
         debug!("Watcher validate taker fee");
-        let validated_f = watcher_ctx
-            .taker_coin
-            .watcher_validate_taker_fee(WatcherValidateTakerFeeInput {
```

Comment:
I think this watcher_validate_taker_fee fn used to fail because inside there is a check that tx.height should be over min_block_number. The case when the tx is in mempool (tx.height is None) is processed and validation does not fail.
But tx.height also may be not None and contain value of '-1' (at least in KMD) if the tx is in a forked branch (so tx validation may fail)

Reply:
> But tx.height also may be not None and contain value of '-1' (at least in KMD) if the tx is in a forked branch (so tx validation may fail)

But we use confirmations not height
https://github.com/KomodoPlatform/komodo-defi-framework/blob/f487947c9d8f454969396986f8f259961ce87f6b/mm2src/coins/utxo/utxo_common.rs#L2004-L2006
I think it fails due to the `get_block_count` delay in having the current height of the blockchain, as the watcher can use a different electrum other than the taker. This is why the retries fix this.